src/scripts/main.js

```diff
@@ -1,3 +1,252 @@
+
+setupInputOnce();
+
+function setupInputOnce() {
+  document.addEventListener('keydown', handleInput, { once: true });
+}
+
+function handleInput(e) {
+  switch (e.key) {
+    case 'ArrowUp':
+      move('up');
+      break;
+    case 'ArrowDown':
+      move('down');
+      break;
+    case 'ArrowLeft':
+      move('left');
+      break;
+    case 'ArrowRight':
+      move('right');
+      break;
+    default:
+      setupInputOnce();
+
+      return;
+  }
+
+  setupInputOnce();
+};
+
+const rows = 4;
+const cols = 4;
+
+const cellsGroupedByRow = [];
+
+for (let i = 0; i < rows; i++) {
+  const row = [];
+
+  for (let j = 0; j < cols; j++) {
+    const index = i * cols + j;
+
+    row.push(cells[index]);
+  }
+  cellsGroupedByRow.push(row);
+}
+
+const cellsGroupedByReversedRow = cellsGroupedByRow.map(
+  row => [...row].reverse());
+
+const cellsGroupedByColumn = [];
+
+for (let i = 0; i < cols; i++) {
+  const row = [];
+
+  for (let j = 0; j < rows; j++) {
+    const index = j * cols + i;
+
+    row.push(cells[index]);
+  }
+  cellsGroupedByColumn.push(row);
+}
+
+const cellsGroupedByReversedColumn = cellsGroupedByColumn.map(
+  column => [...column].reverse());
+
+function move(direction) {
+  let moved = false;
+  let groupedCells;
+
+  switch (direction) {
+    case 'up':
+      groupedCells = cellsGroupedByColumn;
+      break;
+    case 'down':
+      groupedCells = cellsGroupedByReversedColumn;
+      break;
+    case 'left':
+      groupedCells = cellsGroupedByRow;
+      break;
+    case 'right':
+      groupedCells = cellsGroupedByReversedRow;
+      break;
+  }
+
+  for (let i = 0; i < 4; i++) {
+    if (slideTiles(groupedCells)) {
+      moved = true;
+    }
+  }
+
+  if (moved) {
+    setRandomEmptyCell();
+  }
+
+  getScore();
+  setStyles();
+  resetMergedFlags();
+}
+
+function resetMergedFlags() {
+  for (const cell of cells) {
+    cell.merged = false;
+  }
+}
+
+function slideTiles(groupedCells) {
+  let moved = false;
+
+  groupedCells.forEach(group => {
+    if (slideTilesInGroup(group)) {
+      moved = true;
+    }
+  });
+
+  return moved;
+}
+
+function slideTilesInGroup(group) {
+  let moved = false;
+
+  for (let i = 1; i < group.length; i++) {
+    if (!group[i].textContent) {
+      continue;
+    }
+
+    const cellWithTile = group[i];
+
+    if (group[i].textContent
+      && group[i - 1].textContent
+      && group[i].textContent !== group[i - 1].textContent) {
+      continue;
+    }
+
+    for (let j = i - 1; j >= 0; j--) {
+      const targetCell = group[j];
+
+      if (!targetCell) {
+        continue;
+      }
+
+      if (!targetCell.textContent) {
+        targetCell.textContent = cellWithTile.textContent;
+        cellWithTile.textContent = '';
+        moved = true;
+      } else if (targetCell.textContent === cellWithTile.textContent
+          && !cellWithTile.merged && !targetCell.merged) {
+        targetCell.textContent = +targetCell.textContent
```

Comment:
Destructure `targetCell`
```suggestion
      const { textContent, merged } = targetCell
      if (!textContent) {
        targetCell.textContent = cellWithTile.textContent;
        cellWithTile.textContent = '';
        moved = true;
      } else if (textContent === cellWithTile.textContent
          && !cellWithTile.merged && !merged) {
        targetCell.textContent = +textContent + (+cellWithTile.textContent)
```